### PR TITLE
Added few tests for generated subsystems

### PR DESCRIPTION
**Pull request by honza-kasik**
Depends on #204 , still needs work, this part of code is ready for review, though...

### CODE REVIEW

**Comment by honza-kasik** [comment]
Ready to be merged.

**Comment by rhatlapa** [comment]
This contains also changes from #204. Lets wait with having merged !204, then rebase and review.

**Comment by honza-kasik** [comment]
Ready to be merged!

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        page.navigate();
+    }
+
+    @AfterClass
+    public static void afterClass() throws InterruptedException, TimeoutException, IOException {
+        administration.reloadIfRequired();
+    }
+
+    @Test
+    public void verifyActiveRequests() throws Exception {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java:63]: This name looks more as some internal method name than test name, I suggest renaming it to something like `activeRequestsTest` or something similar.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        ModelNodeResult value = operations.readAttribute(REQUEST_CONTROLLER_SUBSYSTEM_ADDRESS, ACTIVE_REQUESTS,
+                ReadAttributeOption.INCLUDE_DEFAULTS);
+
+        value.assertSuccess();
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java]: You probably wanted rather to use `assertDefinedValue()`

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+        Assert.assertEquals(page.getConfigFragment().edit().text(ACTIVE_REQUESTS), value.stringValue());
+    }
+
+    @Test
+    public void editMaxRequests() throws Exception {
+        int value = 42;
+        ModelNodeResult originalValue = operations.readAttribute(REQUEST_CONTROLLER_SUBSYSTEM_ADDRESS, MAX_REQUESTS,
+                ReadAttributeOption.INCLUDE_DEFAULTS);
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java]: Here should be NOT_INCLUDE_DEFAULTS

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        } finally {
+            operations.writeAttribute(REQUEST_CONTROLLER_SUBSYSTEM_ADDRESS, MAX_REQUESTS, originalValue.value())
+                    .assertSuccess();
+        }
+    }
+
+    @Test
+    public void toggleTrackIndividualEndpoints() throws Exception {
+        ModelNodeResult originalValue = operations.readAttribute(REQUEST_CONTROLLER_SUBSYSTEM_ADDRESS,
+                TRACK_INDIVIDUAL_ENDPOINTS, ReadAttributeOption.INCLUDE_DEFAULTS);
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/requestcontroller/RequestControllerSubsystemTestCase.java]: You should use NOT_INCLUDE_DEFAULTS

File: common/src/main/java/org/jboss/hal/testsuite/page/config/RequestControllerPage.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+
+import org.jboss.hal.testsuite.finder.Application;
+import org.jboss.hal.testsuite.finder.FinderNames;
+import org.jboss.hal.testsuite.finder.FinderNavigation;
+import org.jboss.hal.testsuite.page.ConfigPage;
+import org.jboss.hal.testsuite.page.Navigatable;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+
+public class RequestControllerPage extends ConfigPage implements Navigatable {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/RequestControllerPage.java:14]: Missing javadoc.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/naming/NamingTestCase.java [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        page.getResourceManager()
+                .removeResource(BINDING_TBR_NAME)
+                .confirmAndDismissReloadRequiredMessage();
+
+        new ResourceVerifier(BINDING_TBR_ADDRESS, client)
+                .verifyDoesNotExist();
+    }
+
+    @Test
+    public void verifyRemoteNamingServiceExists() {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/naming/NamingTestCase.java]: Don't use `verify` word in naming of test methods.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/noconfig/NoConfigSubsystemTestCaseAbstract.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import org.jboss.hal.testsuite.util.ConfigUtils;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.NotFoundException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebDriver;
+
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public abstract class NoConfigSubsystemTestCaseAbstract {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/noconfig/NoConfigSubsystemTestCaseAbstract.java:23]: Javadoc please

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/noconfig/NoConfigSubsystemTestCaseAbstract.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Drone
+    private WebDriver browser;
+
+    public void verifySubsystemIsPresentByNavigatingToIt(String subsystemLabel) {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/noconfig/NoConfigSubsystemTestCaseAbstract.java:32]: javadoc please

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/securitymanager/SecurityManagerTestCase.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        WizardWindowWithAdvancedSelectBoxOptions window = page.getResourceManager()
+                .addResource(WizardWindowWithAdvancedSelectBoxOptions.class);
+
+        window.pick(DEFAULT)
+                .clickContinue()
+                .finish();
+
+        new ResourceVerifier(DEFAULT_PERMISSIONS_ADDRESS, client)
+                .verifyExists();
+    }
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/securitymanager/SecurityManagerTestCase.java:116]: Make the tests independent to each other => remove need for sequence.

File: common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithAdvancedSelectBoxOptions.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+import org.jboss.hal.testsuite.util.Console;
+import org.jboss.hal.testsuite.util.PropUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class WizardWindowWithAdvancedSelectBoxOptions extends WizardWindow {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithAdvancedSelectBoxOptions.java:15]: javadoc please

File: common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithAdvancedSelectBoxOptions.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+                .map(WebElement::getText)
+                .collect(Collectors.toList());
+    }
+
+    public WizardWindowWithAdvancedSelectBoxOptions clickContinue() {
+        clickButton("Continue");
+
+        Console.withBrowser(browser).waitUntilFinished();
+        return this;
+    }
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithAdvancedSelectBoxOptions.java:65]: Javadoc for each public method please

File: common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithOptionalFields.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
+package org.jboss.hal.testsuite.fragment.shared.modal;
+
+import org.jboss.arquillian.graphene.Graphene;
+import org.jboss.arquillian.graphene.findby.ByJQuery;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class WizardWindowWithOptionalFields extends WizardWindow {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithOptionalFields.java:11]: javadoc please

File: common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithOptionalFields.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+                .visible();
+    }
+
+    private void makeContentActiveForUserInput() {
+        root.findElement(ByJQuery.selector(".default-window-content")).click();
+    }
+
+    private void maximizeWindow() {
+        root.findElement(By.ByClassName.className("icon-resize-full")).click();
+    }
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithOptionalFields.java]: Javadoc for each public method please

File: common/src/main/java/org/jboss/hal/testsuite/page/config/JsfPage.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
+package org.jboss.hal.testsuite.page.config;
+
+import org.jboss.hal.testsuite.finder.Application;
+import org.jboss.hal.testsuite.finder.FinderNames;
+import org.jboss.hal.testsuite.finder.FinderNavigation;
+import org.jboss.hal.testsuite.page.ConfigPage;
+import org.jboss.hal.testsuite.page.Navigatable;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+
+public class JsfPage extends ConfigPage implements Navigatable {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/JsfPage.java:13]: Javadoc

File: common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithOptionalFields.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+
+import org.jboss.arquillian.graphene.Graphene;
+import org.jboss.arquillian.graphene.findby.ByJQuery;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class WizardWindowWithOptionalFields extends WizardWindow {
+
+    public void openAdvancedOptionsTab() {
+        maximizeWindow();
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/WizardWindowWithOptionalFields.java]: there is needed such hack as maximizing window, interesting.

File: common/src/main/java/org/jboss/hal/testsuite/page/config/NamingPage.java [hunk 16]
```diff
@@ -0,0 +1,25 @@
+package org.jboss.hal.testsuite.page.config;
+
+import org.jboss.hal.testsuite.finder.Application;
+import org.jboss.hal.testsuite.finder.FinderNames;
+import org.jboss.hal.testsuite.finder.FinderNavigation;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+
+public class NamingPage extends TreeNavigationPage {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/NamingPage.java:11]: javadoc

File: common/src/main/java/org/jboss/hal/testsuite/page/config/SecurityManagerPage.java [hunk 17]
```diff
@@ -0,0 +1,24 @@
+package org.jboss.hal.testsuite.page.config;
+
+import org.jboss.hal.testsuite.finder.Application;
+import org.jboss.hal.testsuite.finder.FinderNames;
+import org.jboss.hal.testsuite.finder.FinderNavigation;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+
+public class SecurityManagerPage extends TreeNavigationPage {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/SecurityManagerPage.java:11]: javadoc

**Review by rhatlapa — CHANGES_REQUESTED** [review]
You are forgetting javadocs, please add them at least for all public non test methods and classes. Ideally you can add also javadoc for testcase classes.

**Comment by honza-kasik** [comment]
Fixed Javadoc and few other things, still needs work on `SecurityManagerTestCase`

**Comment by honza-kasik** [comment]
`SecurityManagerTestCase` was fixed, but it needs to be expanded, after https://issues.jboss.org/browse/HAL-1225 is resolved. If CI check goes well this PR can be merged.

**Comment by rhatlapa** [comment]
Please rebase (in order to run checks again, now as part of the checks is also findbugs)

**Review by rhatlapa — APPROVED** [review]

**Comment by honza-kasik** [comment]
Rebased, all checks passed.

**Comment by rhatlapa** [comment]
Thanks, merging.

**Pull request merged**